scripts/linkml_to_flat_synapse_jsonschema.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python3
+"""Convert LinkML YAML to Synapse-compatible flat JSON Schema.
+
+This script converts LinkML YAML files to JSON Schema format that is compatible
+with Synapse's JSON Schema service. It handles flattening, version conversion,
+and Synapse-specific formatting requirements.
+"""
+from pathlib import Path
+from typing import Any, Union
+import argparse
+import json
+import subprocess
+import sys
+
+import jsonref
+from linkml.generators.jsonschemagen import JsonSchemaGenerator
+
+
+def run_gen_json_schema(linkml_yaml: str, class_name: str, tmp_json: str) -> None:
+    """Generate JSON Schema from LinkML YAML using Python library.
+
+    Converts a LinkML YAML file to JSON Schema format using the LinkML
+    JsonSchemaGenerator. The generated schema is written to a temporary file
+    for further processing (flattening, version conversion, etc.).
+
+    Args:
+        linkml_yaml (str): Path to the input LinkML YAML file
+        class_name (str): Name of the top-level class to generate schema for.
+                         If empty string, uses the default class from the schema.
+        tmp_json (str): Path to the temporary JSON file where the generated
+                       schema will be written
+
+    Raises:
+        FileNotFoundError: If the LinkML YAML file cannot be found
+        ValueError: If there's an error in the LinkML schema or generation process
+        OSError: If there's an error writing the output file
+
+    Returns:
+        None: The function writes the generated schema to the specified file
+    """
+    print(f"Generating JSON Schema from {linkml_yaml}")
+
+    # Use the Python library instead of shell command
+    generator = JsonSchemaGenerator(linkml_yaml)
+    if class_name:
+        generator.top_class = class_name
+
+    # Generate the schema
+    schema_str = generator.serialize()
+
+    # Write to temporary file
+    with open(tmp_json, "w") as f:
+        f.write(schema_str)
+
+    print(f"Generated JSON Schema written to {tmp_json}")
+
+
+def _convert_to_plain_dict(obj) -> Union[dict, list, Any]:
```

Comment:
Do you have an example here?  This is kind of interesting, because you're stating that the `obj` that is passed in is either `dict` and/or `list` or something else which I would expect it to actually be a `jsonred.JsonRef` object?